server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/orc/ORCSchemaBuilder.java

```diff
@@ -120,11 +121,21 @@ private static TypeDescription orcTypeFromGreenplumType(ColumnDescriptor columnD
      * @param columnTypeModifiers Greenplum type modifiers
      * @return type description object with the specified precision and scale, if any
      */
-    private static TypeDescription setPrecisionAndScale(TypeDescription typeDescription, Integer[] columnTypeModifiers) {
+    private static TypeDescription setPrecisionAndScale(TypeDescription typeDescription, Integer[] columnTypeModifiers, String columnName) {
         // if precision is not defined in Greenplum for a column, columnTypeModifiers will be null
         if (ArrayUtils.isNotEmpty(columnTypeModifiers)) {
             Integer precision = columnTypeModifiers[0];
             if (precision != null) {
+                // In ORC code, no getter for ORC's MAX_PRECISION but there's a getter for ORC's precision.
+                // By default, ORC precision is set to a hardcoded value 38, which is same as MAX_PRECISION's value
+                // and at this point, typeDescription.getPrecision() is still the default precision 38
+                int maxPrecision = typeDescription.getPrecision();
+                // if precision provided in the table DDL exceeds the max precision, ORC will throw an exception
+                // here we provide a PXF error message containing more details
+                if (precision > maxPrecision) {
+                    throw new UnsupportedTypeException(String.format("Column %s is defined as NUMERIC with precision %d " +
+                            "which exceeds maximum supported precision %d of ORC.", columnName, precision, maxPrecision));
```

Comment:
this limitation is a limitation of ORC itself and not related to PXF profiles. Maybe we could make this a little more clear here: 
```suggestion
                            "which exceeds the maximum supported precision value (%d) for ORC files.", columnName, precision, maxPrecision));
```